client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/multimap/RemoteMultimapCacheImpl.java

```diff
@@ -0,0 +1,195 @@
+import org.infinispan.client.hotrod.impl.multimap.operations.PutKeyValueMultimapOperation;
+import org.infinispan.client.hotrod.impl.multimap.operations.RemoveEntryMultimapOperation;
+import org.infinispan.client.hotrod.impl.multimap.operations.RemoveKeyMultimapOperation;
+import org.infinispan.client.hotrod.logging.Log;
+import org.infinispan.client.hotrod.logging.LogFactory;
+import org.infinispan.client.hotrod.marshall.MarshallerUtil;
+import org.infinispan.client.hotrod.multimap.MetadataCollection;
+import org.infinispan.client.hotrod.multimap.RemoteMultimapCache;
+import org.infinispan.commons.marshall.Marshaller;
+
+/**
+ * Remote implementation of {@link RemoteMultimapCache}
+ *
+ * @author karesti@redhat.com
+ * @since 9.2
+ */
+public class RemoteMultimapCacheImpl<K, V> implements RemoteMultimapCache<K, V> {
+
+   private static final Log log = LogFactory.getLog(RemoteMultimapCacheImpl.class, Log.class);
+   private static final boolean trace = log.isTraceEnabled();
+
+   private RemoteCacheImpl<K, Collection<V>> cache;
+   private final RemoteCacheManager remoteCacheManager;
+   protected MultimapOperationsFactory operationsFactory;
+   private Marshaller marshaller;
+   private int estimateKeySize;
+   private int estimateValueSize;
+   protected long defaultLifespan;
+   protected long defaultMaxIdleTime;
+
+   public void init() {
+      operationsFactory = new MultimapOperationsFactory(remoteCacheManager.getTransportFactory(),
+            cache.getName(),
+            remoteCacheManager.getCodec(),
+            remoteCacheManager.getConfiguration());
+      this.marshaller = remoteCacheManager.getMarshaller();
+      this.estimateKeySize = remoteCacheManager.getConfiguration().keySizeEstimate();
+      this.estimateValueSize = remoteCacheManager.getConfiguration().valueSizeEstimate();
+   }
+
+   public RemoteMultimapCacheImpl(RemoteCacheManager rcm, RemoteCache<K, Collection<V>> cache) {
+      if (trace) {
+         log.tracef("Creating multimap remote cache: %s", cache.getName());
+      }
+      this.cache = (RemoteCacheImpl<K, Collection<V>>) cache;
+      this.remoteCacheManager = rcm;
+   }
+
+   @Override
+   public CompletableFuture<Void> put(K key, V value) {
+      if (trace) {
+         log.tracef("About to add (K,V): (%s, %s) lifespan:%d, maxIdle:%d", key, value, defaultLifespan, defaultMaxIdleTime);
+      }
+      assertRemoteCacheManagerIsStarted();
+      K compatKey = hasCompatibility() ? key : null;
+      byte[] marshallKey = MarshallerUtil.obj2bytes(marshaller, key, true, estimateKeySize, estimateValueSize);
+      byte[] marshallValue = MarshallerUtil.obj2bytes(marshaller, value, false, estimateKeySize, estimateValueSize);
+
+      PutKeyValueMultimapOperation<Void> op = operationsFactory.newPutKeyValueOperation(compatKey,
+            marshallKey, marshallValue, defaultLifespan, MILLISECONDS, defaultMaxIdleTime, MILLISECONDS);
+      return CompletableFuture.runAsync(() -> op.execute());
```

Comment:
you have an executor used for async operation in RemoteCacheManager you should use it!

Reply:
ok !

Reply:
I said Ok but I actually created an issue to deal that

https://issues.jboss.org/browse/ISPN-8583